implementations/monix/src/main/scala/sttp/client/impl/monix/MonixWebSockets.scala

```diff
@@ -30,4 +31,26 @@ object MonixWebSockets {
         .guarantee(ws.close())
     }
   }
+  def fromTextPipe: (String => WebSocketFrame) => MonixStreams.Pipe[WebSocketFrame, WebSocketFrame] =
+    f => fromTextPipeF(_.map(f))
+
+  def fromTextPipeF: MonixStreams.Pipe[String, WebSocketFrame] => MonixStreams.Pipe[WebSocketFrame, WebSocketFrame] =
+    p => p.compose(combinedTextFrames)
+
+  def combinedTextFrames: MonixStreams.Pipe[WebSocketFrame, String] = { input =>
+    Observable
+      .fromTask(
+        input
+          .takeWhileInclusive { case WebSocketFrame.Text(_, finalFragment, _) => !finalFragment }
+          .collect { case text: WebSocketFrame.Text => text }
+          .toListL
```

Comment:
maybe we can rewrite this as stream operations, without compiling the stream over and over again? I suspect this might work for websocket streams as they are backend by a mutable data structure (the queue of incoming messages), but would break e.g. for strict streams